test/gtest/uct/cuda/test_switch_cuda_device.cc

```diff
@@ -143,3 +144,59 @@ UCS_TEST_P(test_switch_cuda_device, detect_mem_type_cuda_fabric)
 #endif
 
 _UCT_MD_INSTANTIATE_TEST_CASE(test_switch_cuda_device, cuda_cpy);
+
+class test_p2p_switch_cuda_device : public uct_p2p_rma_test {
+public:
+    void test_xfer(send_func_t send, size_t length, unsigned flags,
+                   ucs_memory_type_t mem_type) override;
+};
+
+void test_p2p_switch_cuda_device::test_xfer(send_func_t send, size_t length,
+                                            unsigned flags,
+                                            ucs_memory_type_t mem_type)
+{
+    int num_devices;
+    ASSERT_EQ(cudaGetDeviceCount(&num_devices), cudaSuccess);
+
+    if (num_devices < 2) {
+        UCS_TEST_SKIP_R("less than two cuda devices available");
+    }
+
+    uct_p2p_rma_test::test_xfer(send, length, flags, mem_type);
+
+    int current_device;
+    ASSERT_EQ(cudaGetDevice(&current_device), cudaSuccess);
+    ASSERT_EQ(cudaSetDevice((current_device + 1) % num_devices), cudaSuccess);
```

Comment:
instead can we run xfer tests on all avail cuda devices? (up to 2)? instead of adding another test class
maybe even using c++ iterator